compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1958,3 +1958,6 @@ error.worker.send.action.not.allowed.in.lock.statement=\
 
 error.worker.receive.action.not.allowed.in.lock.statement=\
   using receive action within the lock statement is not allowed to prevent possible deadlocks
+
+error.empty.regexp.string.disallowed=\
+  regular expression is not allowed: empty RegExp
```

Comment:
Shall we update the error message to something like `Empty regular expressions are not allowed`.
wdyt?

Reply:
@pcnfernando compared to https://github.com/ballerina-platform/ballerina-lang/pull/39848#discussion_r1134970386, WDYT?

Reply:
ok